app/src/main/java/edu/metrostate/cardealer/functionality/Vehicle_ImportFile.java

```diff
@@ -0,0 +1,180 @@
+            for(String filename : files) {
+                InputStream in;
+                OutputStream out;
+                try {
+                    in = assetManager.open(filename);
+
+                    String outDir = Environment.getExternalStorageDirectory().getAbsolutePath() + "/Documents" ;
+
+                    File outFile = new File(outDir, filename);
+
+                    out = new FileOutputStream(outFile);
+                    copyFile(in, out);
+                    in.close();
+                    out.flush();
+                    out.close();
+                } catch(IOException e) {
+                    Log.e("tag", "Failed to copy asset file: " + filename, e);
+                }
+            }
+        }
+    }
+    private void copyFile(InputStream in, OutputStream out) throws IOException {
+        byte[] buffer = new byte[1024];
+        int read;
+        while((read = in.read(buffer)) != -1){
+            out.write(buffer, 0, read);
+        }
+    }
+
+    public void buttonJsonFile(View view){
+        copyAssets();
+        Intent data = new Intent(Intent.ACTION_GET_CONTENT);
+        data.setType("*/*");
+        data = Intent.createChooser(data, "Choose a File");
+        jsonActivityResultLauncher.launch(data);
+    }
+    public void buttonXmlFile(View view){
+        copyAssets();
+        Intent data = new Intent(Intent.ACTION_OPEN_DOCUMENT);
+        data.setType("*/*");
+        data = Intent.createChooser(data, "Choose a File");
+        xmlActivityResultLauncher.launch(data);
+
+    }
+
+    ActivityResultLauncher<Intent> jsonActivityResultLauncher = registerForActivityResult(
+            new ActivityResultContracts.StartActivityForResult(),
+            new ActivityResultCallback<ActivityResult>() {
+                @Override
+                public void onActivityResult(ActivityResult result) {
+
+                    if(result.getResultCode() == Activity.RESULT_OK){
+
+                        Intent data = result.getData();
+
+
+                        Uri uri = null;
+                        if (data != null) {
+                            uri = data.getData();
+                        }
+                        File file = null;
+                        if (uri != null) {
+                            file = new File(uri.getPath());
+                        }
+                        path = null;
+                        if (file != null) {
+                            path = file.getAbsolutePath();
+                        }
+
+                        if (path != null) {
+                            if(path.substring(path.lastIndexOf(".") + 1, path.length()).equals("json"))
+                            {
+                                final TextView jsonField = findViewById(R.id.json_path);
+                                jsonField.setText(path);
+
+                                vehicles = (ArrayList<Vehicle>) VehicleJSONParser.read(file);
+                            }else{
```

Comment:
Completed. Not sure what this is " display which dealers could not add any new vehicles. (Acquisition disabled)."

Reply:
Basically the purpose of this is to show the user what has happened to their file, because what the user sees is that their file is on the screen but not that anything has happened to it. The dialog is there to communicate with the user what has happened.